File: Python/2215.py
class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        res = []

        diff1 = self.find_diff_helper(nums1, nums2)
        res.append(diff1)

        diff2 = self.find_diff_helper(nums2, nums1)
        res.append(diff2)

        return res

    def find_diff_helper(self, nums1, nums2):
        vals_set = set(nums1)
        for num in nums2:
            if num in vals_set:
                vals_set.remove(num)

        return list(vals_set)

File: Java/2215.java
class Solution {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        List<List<Integer>> res = new ArrayList<>();
        List<Integer> diff1 = findDiffHelper(nums1, nums2);
        res.add(diff1);

        List<Integer> diff2 = findDiffHelper(nums2, nums1);
        res.add(diff2);

        return res;
    }
    
    private List<Integer> findDiffHelper(int[] nums1, int[] nums2) {
        Set<Integer> numsSet = Arrays.stream(nums1).boxed().collect(Collectors.toSet());
        for (int num : nums2) {
            if (numsSet.contains(num)) {
                numsSet.remove(num);
            }
        }

        List<Integer> res = new ArrayList<>();
        res.addAll(numsSet);
        return res;
    }
}